openedx/core/djangoapps/user_api/management/commands/populate_retirement_states.py

```diff
@@ -0,0 +1,123 @@
+"""
+Take the list of states from settings.RETIREMENT_STATES and forces the
+RetirementState table to mirror it.
+
+We use a foreign keyed table for this instead of just using the settings
+directly to generate a `choices` tuple for the model because the states
+need to be configurable by open source partners and modifying the
+`choices` for a model field causes new migrations to be generated,
+with a variety of unpleasant follow-on effects for the partner when
+upgrading the model at a later date.
+"""
+from __future__ import print_function
+
+import copy
+import logging
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+from openedx.core.djangoapps.user_api.models import RetirementState, UserRetirementStatus
+
+
+LOGGER = logging.getLogger(__name__)
+
+START_STATE = 'PENDING'
+END_STATES = ['ERRORED', 'ABORTED', 'COMPLETE']
+REQUIRED_STATES = copy.deepcopy(END_STATES)
+REQUIRED_STATES.insert(0, START_STATE)
+REQ_STR = ','.join(REQUIRED_STATES)
+
+
+class Command(BaseCommand):
+    """
+    Implementation of the populate command
+    """
+    help = 'Populates the RetirementState table with the states present in settings.'
+
+    def _validate_new_states(self, new_states):
+        """
+        Check settings for existence of states, required states
+        """
+        if not new_states:
+            raise CommandError('settings.RETIREMENT_STATES does not exist or is empty.')
+
+        if not set(REQUIRED_STATES).issubset(set(new_states)):
+            print(new_states)
```

Comment:
Did you mean to leave this?

Reply:
I did, it proved handy debugging the test so I figured it would help people running the command. I'll add another print with some context about it, though, to make it obvious what it's printing.